editor/src/components/inspector/sections/component-section/data-picker-popup.tsx

```diff
@@ -121,37 +150,37 @@ interface ValueRowProps {
 function ValueRow({ variableOption, idx, onTweakProperty }: ValueRowProps) {
   const colorTheme = useColorTheme()
   const [selectedIndex, setSelectedIndex] = React.useState<number>(0)
-  const childrenLength = variableOption.variableChildren?.length ?? 0
+
+  const childrenLength = variableOption.type === 'array' ? variableOption.children.length : 0
   const [childrenOpen, setChildrenOpen] = React.useState<boolean>(
```

Comment:
should the default value here take into account if there are `matches` inside the object?